### PR TITLE
Homework7

### PR DESCRIPTION
**Pull request by ermekbaeva**
homework7 done

### CODE REVIEW

**Review by cold-flame — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw7/tests/JdiPageTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        metalsAndColorsForm.selectEvenNumbers(String.valueOf(data.getSummary().get(1)));
+        metalsAndColorsForm.clickCalculateButton();
+        metalsAndColorsForm.selectElements(data.getElements());
+        metalsAndColorsForm.selectColor(data.getColor());
+        metalsAndColorsForm.selectMetal(data.getMetals());
+        metalsAndColorsForm.unselectVegetables();
+        metalsAndColorsForm.selectVegetables(data.getVegetables());
+        metalsAndColorsForm.clickSubmitButton();
+
+        assertThat(summaryResult.getText()).containsIgnoringCase(String.valueOf(data.getSum()));
```
[the remark this answers is not in the document]
Comment (ermekbaeva) [reply src/test/java/com/epam/tc/hw7/tests/JdiPageTest.java]: Custom verify was used

File: src/test/java/com/epam/tc/hw7/TestInit.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+
+
+public interface TestInit {
+    @BeforeSuite(alwaysRun = true)
+    static void setUp() {
+        initElements(JdiSite.class);
+        openSite();
+        login();
```
[the remark this answers is not in the document]
Comment (ermekbaeva) [reply src/test/java/com/epam/tc/hw7/TestInit.java]: parameter was added in method login()

File: src/main/java/com/epam/tc/hw7/sections/MetalsAndColorsForm.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            expand = ".caret")
+    public static Dropdown vegetables;
+
+    public void selectVegetables(List<String> value) {
+        for (String val : value) {
+            vegetables.select(val);
+        }
+    }
+
+    public void selectElements(List<String> value) {
```
[the remark this answers is not in the document]
Comment (ermekbaeva) [reply src/main/java/com/epam/tc/hw7/sections/MetalsAndColorsForm.java]: this part was fixed

File: src/main/java/com/epam/tc/hw7/pages/JdiSite.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import com.epam.tc.hw7.sections.MetalsAndColorsForm;
+import org.openqa.selenium.WebElement;
+
+public class JdiSite {
+    public static HomePage homePage;
+    public static LoginForm loginForm;
+    public static MetalsAndColorsPage metalsAndColorsPage;
+    public static MetalsAndColorsForm metalsAndColorsForm;
+    @UI("#submit-button") public static Button submitButton;
+    @UI("img#user-icon") public static Link userIcon;
```
[the remark this answers is not in the document]
Comment (ermekbaeva) [reply src/main/java/com/epam/tc/hw7/pages/JdiSite.java]: remove all variables to HomePage and MetalColorsPage

File: src/main/java/com/epam/tc/hw7/pages/JdiSite.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import com.epam.tc.hw7.sections.LoginForm;
+import com.epam.tc.hw7.sections.MetalsAndColorsForm;
+import org.openqa.selenium.WebElement;
+
+public class JdiSite {
+    public static HomePage homePage;
+    public static LoginForm loginForm;
+    public static MetalsAndColorsPage metalsAndColorsPage;
+    public static MetalsAndColorsForm metalsAndColorsForm;
+    @UI("#submit-button") public static Button submitButton;
```
[the remark this answers is not in the document]
Comment (ermekbaeva) [reply src/main/java/com/epam/tc/hw7/pages/JdiSite.java]: this part was fixed

**Review by cold-flame — APPROVED** [review]

**Pull request merged**